src/components/GenreMoviesList/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { useParams } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+
+import { UseFetch } from '../../hooks/UseFetch';
+import { GET_MOVIES, GET_GENRE } from '../../utils/API/API_ROUTES';
+import { IMovieData } from '../../types/ApiReturn';
+import { MoviesList } from '../MoviesList';
+export const GenreMoviesList = () => {
+  const { genre } = useParams();
+  const [moviesOfGenre, setMoviesOfGenre] = useState<IMovieData[] | null>(null);
+  const [genresIds, SetGenresIds] = useState(0);
+  const { request, loading } = UseFetch();
+  const { url } = GET_MOVIES(1);
+  interface IGenre {
+    id: number;
+    name: string;
+  }
+
+  useEffect(() => {
+    const { url } = GET_GENRE();
+
+    void request(url)
```

Comment:
man it does not make sense cause you can filter movies by query params... But i saw the API doc and to filter by genre you need to send a POST to `/discover/movie` and the payload is weird
```cURl
curl 'https://www.themoviedb.org/discover/movie' \
  -H 'authority: www.themoviedb.org' \
  -H 'accept: text/html, */*; q=0.01' \
  -H 'accept-language: en-BR,en;q=0.9,pt-BR;q=0.8,pt;q=0.7,en-GB;q=0.6,en-US;q=0.5' \
  -H 'content-type: application/x-www-form-urlencoded; charset=UTF-8' \
  -H 'cookie: _ga=GA1.2.983548559.1666386231; tmdb.prefs=%7B%22adult%22%3Afalse%2C%22i18n_fallback_language%22%3A%22en-US%22%2C%22locale%22%3A%22pt-BR%22%2C%22country_code%22%3A%22BR%22%2C%22timezone%22%3A%22America%2FSao_Paulo%22%7D; tmdb.session=AXQloXQNfITM0bnIcQjEXuZmpWJ8tW2v3tQWDa6zogZxliE1wF7btapxnbaNTRUcF6U54rAidmCHbACIEskRbXShaTZovuaerrmR0WFg1tUFiPNnlIEn0YLddfkhuCqD5ocVsTD9L5fETmIJPIZzaP81JyHYqL6SYZlCfyp8HsDE0yZSsya6ieB9i1qNiq3aOzza8Ub2z6IKqKHlDj031X87QZWjHlH827_Inu2d9Z8u; _gid=GA1.2.963445949.1672792893; _dc_gtm_UA-2087971-10=1; _gali=with_genres' \
  -H 'dnt: 1' \
  -H 'origin: https://www.themoviedb.org' \
  -H 'referer: https://www.themoviedb.org/movie' \
  -H 'sec-ch-ua: "Not?A_Brand";v="8", "Chromium";v="108", "Google Chrome";v="108"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36' \
  -H 'x-requested-with: XMLHttpRequest' \
  --data-raw 'air_date.gte=&air_date.lte=2023-07-04&certification=&certification_country=BR&debug=&first_air_date.gte=&first_air_date.lte=&ott_region=BR&page=1&primary_release_date.gte=&primary_release_date.lte=&region=&release_date.gte=&release_date.lte=2023-07-04&show_me=0&sort_by=popularity.desc&vote_average.gte=0&vote_average.lte=10&vote_count.gte=0&with_genres=12&with_keywords=&with_networks=&with_origin_country=&with_original_language=&with_ott_monetization_types=&with_ott_providers=&with_release_type=&with_runtime.gte=0&with_runtime.lte=400' \
  --compressed
```

well, you'll send genre ID in `with_genres` and you'll have all movies about that genre.